v23.1/performance-recipes.md

```diff
@@ -77,15 +84,113 @@ This section provides solutions for common performance issues in your applicatio
+  - The details of the **blocking** transaction, directly below the **Contention Insights** section. Click the blocking transaction to view its details.
+
+##### Cancel a blocking transaction
+
+1. [Identify the **blocking** transaction](#identify-conflicting-transactions) and view its transaction execution details.
+1. Click its **Session ID** to open the **Session Details** page.
+  <img src="{{ 'images/v23.1/ui-sessions-details-page.png' | relative_url }}" alt="Sessions Details Page" style="border:1px solid #eee;max-width:100%" />
+1. Click **Cancel Statement** to cancel the **Most Recent Statement** and thus the transaction, or click **Cancel Session** to cancel the session issuing the transaction.
+
+##### Identify transactions and objects that experienced lock contention
+
+To identify transactions that experienced [lock contention](performance-best-practices-overview.html#transaction-contention) in the past:
+
+- In the **Transaction Executions** view on the **Insights** page ([{{ site.data.products.db }} Console](../cockroachcloud/insights-page.html#transaction-executions-view) and [DB Console](ui-insights-page.html#transaction-executions-view), look for a transaction with the **High Contention** insight. Click the transaction's execution ID and view the transaction execution details, including the details of the blocking transaction.
+- Visit the **Transactions** page ([{{ site.data.products.db }} Console](../cockroachcloud/transactions-page.html) and [DB Console](ui-transactions-page.html)) and sort transactions by **Contention Time**.
+
+To view tables and indexes that experienced [contention](performance-best-practices-overview.html#transaction-contention):
+
+- Query the [`crdb_internal.transaction_contention_events`](crdb-internal.html#transaction_contention_events) table to view [transactions that have blocked other transactions](crdb-internal.html#transaction-contention-example).
+- Query the [`crdb_internal.cluster_contended_tables`](crdb-internal.html#cluster_contended_tables) table to [view all tables that have experienced contention](crdb-internal.html#view-all-tables-that-have-experienced-contention).
+- Query the [`crdb_internal.cluster_contended_indexes`](crdb-internal.html#cluster_contended_indexes) table to [view all indexes that have experienced contention](crdb-internal.html#view-all-indexes-that-have-experienced-contention).
+- Query the [`crdb_internal.cluster_contention_events`](crdb-internal.html#cluster_contention_events) table 
+to [view the tables, indexes, and transactions with the most time under contention](crdb-internal.html#view-the-tables-indexes-with-the-most-time-under-contention).
+
+##### Reduce transaction contention
+
+Because [contention](performance-best-practices-overview.html#transaction-contention) affects your cluster performance, and is often reported after it has already resolved, preventing contention is highly effective:
```

Comment:
Perhaps "highly effective" needs to change to "more effective"? I think I'm thrown by the first clause — does contention affecting your cluster performance have anything to do with the fact that it is report after the fact? Here is a suggestion, which I hope captures your meaning here:

> Contention is often reported after it has already resolved, therefore, preventing contention before it affects your cluster's performance is a more effective approach. The following describes ways to (reduce)/(prevent) transaction contention:

Reply:
Thank you! This is a big improvement.